### ISSUE TITLE
rock/edge damage not always correct 

### ISSUE THREAD

**Issue by patscho** [captain-dread/pillage-helper-web#1]
I haven't fully figured this out yet but when I sail with a sloop and I'm battling a dhow or other ship with medium/large cannon balls the rock/edge damage is not calculated correctly. For example sloop vs dhow. One rock bump should be 0.5 a small cannon ball or 8.33% The helper will see it as 0.5 medium or 12.5% cannon ball which is not correct. It gets even worse with ships that shoot large cannon balls.

**Comment by captain-dread** [comment]
> I haven't fully figured this out yet but when I sail with a sloop and I'm battling a dhow or other ship with medium/large cannon balls the rock/edge damage is not calculated correctly. For example sloop vs dhow. One rock bump should be 0.5 a small cannon ball or 8.33% The helper will see it as 0.5 medium or 12.5% cannon ball which is not correct. It gets even worse with ships that shoot large cannon balls.

Thanks for letting me know! Pushed an update today that should have fixed this. Let me know if it's still not calculating correctly.

**Comment by patscho** [comment]
Still same behavior. I added 3 screenshots. Each of them of a sloop with damage taken from one click on the edge/rock damage button. They all should have the same percentage of damage but as you can see that is not the case. I took a look at your code although JavaScript is not really my thing. For some reason when fighting a ship with bigger shots the percentage isn't taken into account correctly for rock/edge damage and maybe also for ram damage. As you can see here: https://yppedia.puzzlepirates.com/Ram the rock damage for a sloop is the equivalent of 0.5 small cannon ball, 0.333 medium cannon ball or 0.25 large cannon ball. For some reason the helper sticks to 0.5 regardless of the shot size. When the rest of the damage is calculate with a different shot size then things start to go wrong. I hope this helps in finding the root cause and fixing this issue. 

![Schermafbeelding 2024-01-01 222911](https://github.com/captain-dread/pillage-helper-web/assets/18365617/7d682d2c-f005-42b2-85f7-0c83b0c84ddd)
![Schermafbeelding 2024-01-01 222951](https://github.com/captain-dread/pillage-helper-web/assets/18365617/f5e0c8f6-12c3-4267-892c-bb9924e3da68)
![Schermafbeelding 2024-01-01 223022](https://github.com/captain-dread/pillage-helper-web/assets/18365617/a8965e37-f7bb-4ea1-834b-68d887fab15a)

**Comment by captain-dread** [comment]
Thanks for checking it again. I'm going to take a deeper dive into this. I thought it might've been just a wrong variable thing. I originally wrote it in python then translated it into javascript, so I must've missed something.

**Comment by captain-dread** [comment]
> Still same behavior. I added 3 screenshots. Each of them of a sloop with damage taken from one click on the edge/rock damage button. They all should have the same percentage of damage but as you can see that is not the case. I took a look at your code although JavaScript is not really my thing. For some reason when fighting a ship with bigger shots the percentage isn't taken into account correctly for rock/edge damage and maybe also for ram damage. As you can see here: https://yppedia.puzzlepirates.com/Ram the rock damage for a sloop is the equivalent of 0.5 small cannon ball, 0.333 medium cannon ball or 0.25 large cannon ball. For some reason the helper sticks to 0.5 regardless of the shot size. When the rest of the damage is calculate with a different shot size then things start to go wrong. I hope this helps in finding the root cause and fixing this issue.
> 
> ![Schermafbeelding 2024-01-01 222911](https://private-user-images.githubusercontent.com/18365617/293607814-7d682d2c-f005-42b2-85f7-0c83b0c84ddd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDg1NzEsIm5iZiI6MTcwNDQwODI3MSwicGF0aCI6Ii8xODM2NTYxNy8yOTM2MDc4MTQtN2Q2ODJkMmMtZjAwNS00MmIyLTg1ZjctMGM4M2IwYzg0ZGRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTljYWM0NTliZDkzODE1ZTAxMTM2ZmFmYzM0M2Q0NzQxZjhjOGJkNTcyMTljNjc1NjMyN2Y5ZDBhOWY1YTVlYWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4wlwJOdJuEoLAgQKl_t4gA8757CnbkzwSJ8s7aaouck) ![Schermafbeelding 2024-01-01 222951](https://private-user-images.githubusercontent.com/18365617/293607817-f5e0c8f6-12c3-4267-892c-bb9924e3da68.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDg1NzEsIm5iZiI6MTcwNDQwODI3MSwicGF0aCI6Ii8xODM2NTYxNy8yOTM2MDc4MTctZjVlMGM4ZjYtMTJjMy00MjY3LTg5MmMtYmI5OTI0ZTNkYTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhZDEyNWE0ZmE5OWEyZjU3MDJjNWJhODM0MGE5NWRmN2U5YjI4M2U0YzllZmFkZTk4ZmFmY2U4M2QxZmNmNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sRE06LbCxmONYecEreEjahhAA85ic4LhbhjZKlz73K0) ![Schermafbeelding 2024-01-01 223022](https://private-user-images.githubusercontent.com/18365617/293607821-a8965e37-f7bb-4ea1-834b-68d887fab15a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ0MDg1NzEsIm5iZiI6MTcwNDQwODI3MSwicGF0aCI6Ii8xODM2NTYxNy8yOTM2MDc4MjEtYTg5NjVlMzctZjdiYi00ZWExLTgzNGItNjhkODg3ZmFiMTVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA0VDIyNDQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ5MWVhMzI5ZDc0ZmE2YjRmMmVlOWY0NGVjZWRiOGMwOTQ5ODljYzc2YmRmZWNhOGEyYjY2MDg0Mjc0M2UxNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.w1vM5fxvkiSNYqJi7z2rYryAv7NyOLrHwOW4Lfm5Crs)

I have pushed out an update that should fix this. It was calculating the rock damage using it's own ship cannon size, and never the based on the enemies cannon size. I missed this while translating the code over from python. Let me know if you still see this issue on your end

**Comment by patscho** [comment]
Yes! It's working correctly now. Thanks a lot also on behalf of other
people I know that use this nice tool!

Kind regards,

Patrick
